dvc/repo/gc.py

```diff
@@ -46,7 +46,6 @@ def gc(
         all_tags=all_tags,
         all_commits=all_commits,
         all_branches=all_branches,
-        cloud=cloud,
```

Comment:
Not sure why we are removing it here, while we still have `if cloud` down below. Am i missing something?

Reply:
Previously, if we just did `-c` or `cloud=True`, it'd not raise a warning and by default assume `--workspace`. Now, even if `-c` is specified, unless `workspace/all_tags/all_commits/all_branches` are also specified, it will raise warning.

Reply:
@skshetry Looks like there is some misunderstanding here. I simply don't see `cloud` defined anywhere in this method, while it is used down below in `if cloud`.

Reply:
So, `_raise_error_if_all_disabled` would throw following error if all of the arguments were false. 
```sh
ERROR: unexpected error - Invalid Arguments. Either of (all_branches, all_tags, all_commits, workspace) needs to be enabled.
```

So, when we do, `dvc gc -c`, `cloud` is `True`, and hence would not have thrown any error, and simply worked as if `-cw` was specified (as default is `workspace`).

But, now,I have removed `cloud` from the arguments. Even if `cloud` is set to `True`, unless any of the other arguments are `True`, it will raise an error. So: `dvc gc -c` will now throw the same error.

Reply:
Are you confusing it with passing to the `repo.gc()` function?

Reply:
Ok, there was a misunderstanding on my side. I thought you are removing `cloud` from method's(gc) arguments, while you are simply removing it from arguments for `    _raise_error_if_all_disabled(	` . :ok_hand:

Reply:
Ahh, I see were the confusion comes from:
![Screenshot from 2020-03-17 18-23-51](https://user-images.githubusercontent.com/18718008/76856968-8e3fca00-687c-11ea-9299-2838fbbe6e02.png)